nucypher/cli/painting.py

```diff
@@ -854,41 +855,47 @@ def echo_solidity_version(ctx, param, value):
 
 def paint_worklock_status(emitter, registry: BaseContractRegistry):
     from maya import MayaDT
-    worklock_agent = ContractAgency.get_agent(WorkLockAgent, registry=registry)
+    worklock_agent = ContractAgency.get_agent(WorkLockAgent, registry=registry)  # type: WorkLockAgent
     blockchain = worklock_agent.blockchain
 
-    # Agency
-    token_agent = ContractAgency.get_agent(NucypherTokenAgent, registry=registry)
-
     # Time
-    start = MayaDT(worklock_agent.contract.functions.startBidDate().call())
-    end = MayaDT(worklock_agent.contract.functions.endBidDate().call())
-    duration = end - start
-    remaining = end - maya.now()
+    bidding_start = MayaDT(worklock_agent.contract.functions.startBidDate().call())
+    bidding_end = MayaDT(worklock_agent.contract.functions.endBidDate().call())
+    cancellation_end = MayaDT(worklock_agent.contract.functions.endCancellationDate().call())
 
-    # TODO: Include calculated refund and deposit rates
+    bidding_duration = bidding_end - bidding_start
+    cancellation_duration = cancellation_end - bidding_start
+    now = maya.now()
+    bidding_remaining = bidding_end - now if bidding_end > now else timedelta()
+    cancellation_remaining = cancellation_end - now if cancellation_end > now else timedelta()
 
     payload = f"""
 
 Time
 ======================================================
-Start Date ........ {start}
-End Date .......... {end}
-Duration .......... {duration}
-Time Remaining .... {remaining} 
+Bidding Start Date ................... {bidding_start}
+Bidding End Date ..................... {bidding_end}
+Cancellation Window End Date ......... {cancellation_end}
+Bidding Duration ..................... {bidding_duration}
+Cancellation Window Duration ......... {cancellation_duration}
```

Comment:
These two lines feel redundant since we have the start dates, the end dates, and the time remaining...

Reply:
I'm not sure,  as for me - more information is better. But it's not very strong opinion.  What others think?

Reply:
Inclusion of this data is useful, IMO.

Reply:
👍